packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -412,8 +412,16 @@ export class IndexedDbPersistence implements Persistence {
       })
       .catch(e => {
         if (!this.allowTabSynchronization) {
-          throw e;
-        }
+          if (e.name === 'IndexedDbTransactionError') {
+            logDebug(LOG_TAG, "Failed to extend owner lease: ", e);
+            // Proceed in primary mode since the client was not initialized
```

Comment:
Why is proceeding here assuming we're the primary a useful thing?

When this method is called in the course of `enablePersistence`, don't we want to report a failure to open the database?

Reply:
The idea was to let start() proceed if this check failed. I have now changed the code to fail start() and hence enablePersistence() if we get an IndexedDbTransactionError and `synchronizeTabs` is false. That may however hide that we can never obtain the lease. The fix for this was pretty straightforward, but I ended up changing the callsites to use `this.runTransaction` instead of `this.simpleDb.runTransaction`, as this allows me to use MockPersistence to inject failures.

Note that the behavior is not all that different: https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/local/indexeddb_persistence.ts#L808

We make one additional call to verify the lease (which succeeds if no other client has exclusive access). We also obtain access to all ObjectStores, but I think that is not all that different since there is no operation that doesn't also access the DbPrimaryStore.